sdk/src/test/unit/java/com/finbourne/features/FeatureExtractorTest.java

```diff
@@ -0,0 +1,45 @@
+package com.finbourne.features;
+
+import org.apache.commons.codec.Charsets;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.assertThat;
+
+public class FeatureExtractorTest {
```

Comment:
Add a few more tests here:

1. methods with no annotation
2. method with empty annotation `LusidFeature("")`
3. duplicate annotations i.e. same annotation value on multiple tests

Reply:
to allow unit testing for this, added new custom exceptions with custom error messages, as well as dummy files under com.finbourne.features.dummyfiles namespace